File: backend/FleetMind.Api/Models/Cargo.cs
using System;
using FleetMind.Api.Common.Constants;
using FleetMind.Api.Models.Common;

namespace FleetMind.Api.Models
{
    public class Cargo : BaseEntity
    {
        public Guid VoyageId { get; set; }
        public Voyage Voyage { get; set; } = null!;

        public string Description { get; set; } = string.Empty;
        public string Type { get; set; } = string.Empty;
        public string Status { get; set; } = CargoStatus.Pending;
        
        public decimal WeightKg { get; set; }
        public decimal DeclaredValue { get; set; }
        
        public string ConsigneeName { get; set; } = string.Empty;
        public string? HazardNotes { get; set; }
    }
}

File: frontend/src/types/cargo.ts
import { PaginationQuery } from './pagination';

export interface Cargo {
  id: string;
  voyageId: string;
  voyageNumber: string | null;
  description: string;
  type: string;
  status: string;
  weightKg: number;
  declaredValue: number;
  consigneeName: string;
  hazardNotes: string | null;
  createdAt: string;
  warnings: string[];
}

export interface CreateCargoPayload {
  voyageId: string;
  description: string;
  type: string;
  weightKg: number;
  declaredValue: number;
  consigneeName: string;
  hazardNotes?: string;
}

export interface UpdateCargoPayload {
  description: string;
  status: string;
  weightKg: number;
  declaredValue: number;
  consigneeName: string;
  hazardNotes?: string;
}

export interface CargoQuery extends PaginationQuery {
  voyageId?: string;
  status?: string;
  type?: string;
}
